.github/workflows/retry-worfklow-run.yml

```diff
@@ -81,14 +92,27 @@ jobs:
           owner: ${{ inputs.owner }}
           repositories: ${{ inputs.repository }}
 
-      - name: Wait for worfklow run-id=${{ inputs.run-id }} to complete, repo=${{ inputs.owner }}/${{ inputs.repository }}
+      - name: Wait for workflow run-id=${{ inputs.run-id }} to complete, repo=${{ inputs.owner }}/${{ inputs.repository }}
         env:
+          ATTEMPT: ${{ inputs.attempt }}
           GH_REPO: ${{ inputs.owner }}/${{ inputs.repository }}
           GH_TOKEN: ${{ steps.github-token.outputs.token }}
           RUN_ID: ${{ inputs.run-id }}
         run: |
+          # Wait until the workflow run is completed
+
           echo "Waiting for the workflow run to complete ..."
-          gh run watch "${RUN_ID}"
+          ATTEMPT_GH_CLI_FLAG="${ATTEMPT:+--attempt $ATTEMPT}"
+          while true; do
```

Comment:
I would avoid any `while true` loops.
They swallow the condition somewhere in the body.
Instead i would go for `while [ "$status" != "completed" ]`